test/plugins.spec.js

```diff
@@ -114,7 +114,7 @@ test('add options, merging correctly', async () => {
   });
 });
 
-test('do not allow collisions in smartContract.addContracts', async () => {
+test('override duplicates in smartContract.addContracts with the new address', async () => {
```

Comment:
please clean up this test by removing `bar` and `baz` since they're no longer relevant to the test's new purpose

Reply:
can you also change this test to have two plugins, each of which overrides the same contract with a different address, and verify that the last one in takes precedence--that is a key part of the functionality and so we should reflect it in the test